src/goto-instrument/contracts/contracts.cpp

```diff
@@ -1124,7 +1124,7 @@ void code_contractst::apply_loop_contract(
       {
         do
           i++;
-        while(i->is_dead());
+        while(i < loop_end && i->is_dead());
```

Comment:
You're now introducing another instance of the comparison you intended to avoid.

Reply:
Oops!! Good catch. Thanks.